### PR TITLE
test: track payload size for cli-hello-world-ivy

### PR DESCRIPTION
**Pull request by cexbrayat**
Adds size bundle tracking to the integration test `cli-hello-world-ivy` and updates the app itself to reflect the result of using the  `experimentalIvy` flag on the CLI.
The result is similar to:

    npx @angular/cli@next new cli-hello-world-ivy --experimental-ivy --defaults

But replaces the current (cli `7.2.0-rc.0`) `renderComponent` bootstrap with the usual `platformBrowserDynamic` one.
It also keeps what the app did (display a pipe, tests it).

This has been added to track regressions in bundle sizes of a CLI app with Ivy enabled.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe:

Updates an integration test and adds tracking for the bundle size of the CLI app with Ivy enabled (which is currently weirdly high at 496K).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This has been discussed on Slack with @petebacondarwin @IgorMinar and @alan-agius4

### CODE REVIEW

File: integration/_payload-limits.json [hunk 1]
```diff
@@ -8,6 +8,15 @@
...
   },
+  "cli-hello-world-ivy": {
+    "master": {
+      "uncompressed": {
+        "runtime": 1440,
+        "main": 507677,
+        "polyfills": 38390
+      }
+    }
+  },
```
Comment (cexbrayat) [inline integration/_payload-limits.json]: I hacked the `payload-size.sh` script to make it run on macOS, so I'm not 100% confident in the sizes.
I'm also not sure if runtime and polyfills is relevant...

File: integration/cli-hello-world-ivy/src/index.html [hunk 2]
```diff
@@ -2,12 +2,11 @@
 <html lang="en">
 <head>
   <meta charset="utf-8">
-  <title>CliHelloWorld</title>
+  <title>CliHelloWorldIvy</title>
   <base href="/">
 
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
-  <script>window['ngDevMode'] = true;</script>
```
Comment (cexbrayat) [inline integration/cli-hello-world-ivy/src/index.html]: I don't think this is useful anymore, but maybe I'm mistaken.

Comment (petebacondarwin) [reply integration/cli-hello-world-ivy/src/index.html]: I believe that this is now handled by CLI

**Comment by JoostK** [comment]
Nice! The current payload size is so high because `platformBrowserDynamic` pulls in a lot of ViewEngine together with the Compiler infrastructure. If you switch to `platformBrowser` it will become significantly smaller (but still quite large). I wonder if the CLI replaces `platformBrowserDynamic` with `platformBrowser` during prod builds? Is the prod build actually tested here?

**Comment by cexbrayat** [comment]
@JoostK thanks! Yes it should be the prod build. I don't know where the issue is, but we indeed end up with the compiler in the bundle (discussion in `ivy-cli-build-size` on Slack).

File: integration/cli-hello-world-ivy/src/tsconfig.app.json [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
     "types": []
   },
   "exclude": [
     "test.ts",
     "**/*.spec.ts"
-  ]
+  ],
+  "angularCompilerOptions": {
+    "enableIvy": "ngtsc"
+  }
```
Comment (gkalpak) [inline integration/cli-hello-world-ivy/src/tsconfig.app.json]: Does moving this here form `tsconfig.json` make a difference?

**Comment by JoostK** [comment]
@cexbrayat It may be a good idea to verify that native ngtsc behaves the same. The `cli-hello-world-ivy` app uses ngcc to convert the `packages_dist` output, perhaps there's currently some discrepancies between its output when compared to native ngtsc builds.

**Review by gkalpak — COMMENTED** [review]
Thx for the updates :tada:

Basically, this PR contains two changes (that should be in two separate commits imo):
1. Update the `cli-hello-world-ivy` integration project to latest CLI. 
2. Enable payload-size checking for the same project.

The former looks great. I am not convinced about the latter at this point.
What is the reasoning for adding this check now, given that Ivy is still not finalized and bundle sizes are may change (hopefully downwards).

File: integration/cli-hello-world-ivy/src/tsconfig.app.json (hunk 3 above)
Comment (cexbrayat) [reply integration/cli-hello-world-ivy/src/tsconfig.app.json]: I don't think it really does, and this how `experimentalIvy` currently generates it

**Comment by cexbrayat** [comment]
@gkalpak I can split in two commits if needed sure.

We added the size tracking because the result is currently weird (the bundle contains the compiler) after discussing it with Igor on Slack

**Comment by gkalpak** [comment]
Also note that BuildOptimizer currently does not work correctly with `platformBrowserDynamic()` (see angular/angular-cli#13261), so as long as the prod build is broken, it makes even less sense to check payload sizes imo.

**Comment by cexbrayat** [comment]
Yeah we found out when digging out this issue :)

**Comment by cexbrayat** [comment]
@gkalpak I splitted into two commits

File: integration/cli-hello-world-ivy/src/tsconfig.app.json (hunk 3 above)
Comment (alan-agius4) [reply integration/cli-hello-world-ivy/src/tsconfig.app.json]: The reason being that it’s here is that you can have multiple projects which don’t necessarily use ivy.

**Review by gkalpak — APPROVED** [review]
The first commit seems more like `build: ...` (rather than `test: ...`). I would also be better to mention the cli version explicitly (e.g. ``build: update `cli-hello-world-ivy` to cli@...``).

##
The second commit message mentions "track payload size", but the change does not really track it (in the sense that it saves it to Firebase and we can see how it evolves over time). It rather checks the payload size and fails the build if it deviates more than 1% percent from the set limits).
You might want to be more explicit on that in the commit message to avoid confusion in the future.

Also, if the intention was to actually record the size changes in Firebase, then you need more changes :smiley:

##
The code LGTM. I am still not convinced it adds much benefit to check the size when there are known bugs that affect bundling, but if @IgorMinar thinks so then it probably is so :grin:

**Comment by cexbrayat** [comment]
@gkalpak thanks for the feedback, I updated the commit messages 👍  I admit that I don't know the build setup well enough to add the necessary for recording the sizes in Firebase, but as you say that's probably not necessary right now.

**Comment by gkalpak** [comment]
(FWIW, I'd be happy to help setting that up, if/when we decide it makes sense to do so.)

**Review by IgorMinar — APPROVED** [review]
This looks great! Thanks so much for getting it done!

**Comment by IgorMinar** [comment]
As far as test/build/ci commit message types goes I'd categorize this as ci or test. See my definition of types at https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

**Comment by cexbrayat** [comment]
@IgorMinar Haha, changing a commit message is not the hardest part ;) I'll revert from `build` to `test`.

**Comment by gkalpak** [comment]
> As far as test/build/ci commit message types goes I'd categorize this as ci or test. See my definition of types at https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

The PR had 2 commits (one updating the project to the latest cli and one updating the tests to check for payload size changes).
The way I understand the type descriptions, I would still categorize the 1st commit as `build` (and the 2nd as `test`).
Is my understandign wrong? Is it because the `cli-hello-world-ivy` project is used for testing purposes, so everything related to that is a `test` change?

(@IgorMinar, asking to make sure we are on the same page and that I use appropriate types in my commit messages.)

**Pull request closed without merging**